### PR TITLE
Replace most of Vavr `Match`/`Case` with Java 12+ `switch` expressions

### PR DESCRIPTION
**Pull request by PawelLipski**
1️⃣.6️⃣ K

### CODE REVIEW

File: backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/aux/CreateGitMacheteRepositoryAux.java [hunk 1]
```diff
@@ -98,14 +94,15 @@ public IGitMacheteRepositorySnapshot createSnapshot(BranchLayout branchLayout) t
...
-    val ongoingOperationType = Match(gitCoreRepository.deriveRepositoryState()).of(
-        Case($(GitCoreRepositoryState.CHERRY_PICK), OngoingRepositoryOperationType.CHERRY_PICKING),
-        Case($(GitCoreRepositoryState.MERGING), OngoingRepositoryOperationType.MERGING),
-        Case($(GitCoreRepositoryState.REBASING), OngoingRepositoryOperationType.REBASING),
-        Case($(GitCoreRepositoryState.REVERTING), OngoingRepositoryOperationType.REVERTING),
-        Case($(GitCoreRepositoryState.APPLYING), OngoingRepositoryOperationType.APPLYING),
-        Case($(GitCoreRepositoryState.BISECTING), OngoingRepositoryOperationType.BISECTING),
-        Case($(), OngoingRepositoryOperationType.NO_OPERATION));
+    val ongoingOperationType = switch (gitCoreRepository.deriveRepositoryState()) {
+      case CHERRY_PICKING -> OngoingRepositoryOperationType.CHERRY_PICKING;
```
Comment (PawelLipski) [inline backend/impl/src/main/java/com/virtuslab/gitmachete/backend/impl/aux/CreateGitMacheteRepositoryAux.java:98]: Renamed the enum value for consistency

File: backend/impl/src/test/java/com/virtuslab/gitmachete/backend/integration/StatusAndDiscoverIntegrationTestSuite.java [hunk 2]
```diff
@@ -176,12 +173,14 @@ private void printCommonParts(IGitMacheteRepositorySnapshot gitMacheteRepository
...
-          Case($(BehindRemote), "behind " + remoteName),
-          Case($(DivergedFromAndNewerThanRemote), "diverged from " + remoteName),
-          Case($(DivergedFromAndOlderThanRemote), "diverged from & older than " + remoteName)));
+      sb.append(switch (syncToRemoteStatus) {
+        case Untracked -> "untracked";
+        case AheadOfRemote -> "ahead of " + remoteName;
+        case BehindRemote -> "behind " + remoteName;
+        case DivergedFromAndNewerThanRemote -> "diverged from " + remoteName;
+        case DivergedFromAndOlderThanRemote -> "diverged from & older than " + remoteName;
+        case InSyncToRemote, NoRemotes -> throw new IllegalStateException("Unexpected value: " + syncToRemoteStatus);
```
Comment (PawelLipski) [inline backend/impl/src/test/java/com/virtuslab/gitmachete/backend/integration/StatusAndDiscoverIntegrationTestSuite.java:182]: To satisfy the exhaustiveness check in compiler

File: backend/impl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveCreatedAndDuplicatedAndSkippedBranchesUnitTestSuite.java [hunk 3]
```diff
@@ -26,6 +27,7 @@
   private IGitMacheteRepositorySnapshot invokeCreateSnapshot(
       BranchLayout branchLayout, IGitCoreLocalBranchSnapshot... localBranchSnapshots) {
     when(gitCoreRepository.deriveAllRemoteNames()).thenReturn(List.empty());
+    when(gitCoreRepository.deriveRepositoryState()).thenReturn(GitCoreRepositoryState.NO_OPERATION);
```
Comment (PawelLipski) [inline backend/impl/src/test/java/com/virtuslab/gitmachete/backend/unit/GitMacheteRepository_deriveCreatedAndDuplicatedAndSkippedBranchesUnitTestSuite.java:30]: It used to return `null`, was passing under old logic, doesn't pass now

File: gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java [hunk 4]
```diff
@@ -598,22 +593,17 @@ public List<IGitCoreCommit> deriveCommitRange(IGitCoreCommit fromInclusive, IGit
...
-        Case($(isIn(RepositoryState.REVERTING, RepositoryState.REVERTING_RESOLVED)),
-            GitCoreRepositoryState.REVERTING),
-        Case($(RepositoryState.APPLY),
-            GitCoreRepositoryState.APPLYING),
-        Case($(RepositoryState.BISECTING),
-            GitCoreRepositoryState.BISECTING),
-        Case($(), GitCoreRepositoryState.NORMAL));
-    // @formatter:on
+    val state = jgitRepoForWorktreeGitDir.getRepositoryState();
+    return switch (state) {
```
Comment (mkondratek) [inline gitCore/jGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java:597]: finally... now it looks like human 👨

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/UnmanagedBranchNotificationFactory.java [hunk 5]
```diff
@@ -127,9 +127,7 @@ private NotificationAction getOpenMacheteFileAction() {
           val dataContext = new DataContext() {
             @Override
             public @Nullable Object getData(String dataId) {
-              return Match(dataId).of(
-                  typeSafeCase(CommonDataKeys.PROJECT, project),
-                  Case($(), (Object) null));
+              return dataId.equals(CommonDataKeys.PROJECT.getName()) ? project : null;
```
Comment (mkondratek) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/UnmanagedBranchNotificationFactory.java:130]: why don't we use `switch` here too? I know, logically it's a bit redundant but imho it's more readable 

and what about the other `DataContext#getData`s we define? Why don't we have switch there? 🤔

**Review by mkondratek — COMMENTED** [review]
questions ❓ ❔ ❓

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/UnmanagedBranchNotificationFactory.java (hunk 5 above)
Comment (PawelLipski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/UnmanagedBranchNotificationFactory.java:130]: `switch` only accepts constant expressions for `case` ☹️ so Vavr `Match` is still left in a few places

**Comment by PawelLipski** [comment]
Draft coz let's wait for #1608 first, there's gonna be some changes around `switch`-intensive `Traverse*` classes

**Pull request merged**